crates/proof-of-sql/src/sql/proof/indexes.rs

```diff
@@ -93,6 +93,10 @@ impl Indexes {
                 if range.is_empty() {
                     Some(Zero::zero())
                 } else {
+                    // Do not run `compute_truncated_lagrange_basis_sum` if the range is too large.
+                    if range.end as usize > 2usize.pow(evaluation_point.len() as u32) {
+                        return None;
+                    }
```

Comment:
@JayWhite2357 The essence of the problem is that `compute_truncated_lagrange_basis_sum(table_length, evaluation_point)` panics if `table_length > 2^evaluation_point.len()` which indeed doesn't naturally happen. However it happens here: https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/main/crates/proof-of-sql/src/sql/proof/verifiable_query_result_test_utility.rs#L127-L130.

Since we want to get rid of `Indexes` anyway since there are no longer sparse `Indexes` ever since #137 I think it is a valid approach. I will remove `return` though. What do you think?